plutus-core/untyped-plutus-core/test/Evaluation/Builtins/Bitwise.hs

```diff
@@ -0,0 +1,1156 @@
+                EvaluationResult (Untyped.Term Name DefaultUni DefaultFun ()))
+demorganing fun fun' x y = do
+  let x' = mkConstant @ByteString () x
+  let y' = mkConstant @ByteString () y
+  let comp = mkIterApp () (builtin () ComplementByteString) [
+        mkIterApp () (builtin () fun) [x', y']
+        ]
+  let comp' = mkIterApp () (builtin () fun') [
+        mkIterApp () (builtin () ComplementByteString) [x'],
+        mkIterApp () (builtin () ComplementByteString) [y']
+        ]
+  bitraverse cekEval cekEval (comp, comp')
+
+data AssociativeCase =
+  AssociativeMismatched ByteString ByteString ByteString |
+  AssociativeMatched ByteString ByteString ByteString ByteString
+  deriving stock (Eq, Show)
+
+getAssociativeArgs :: AssociativeCase -> (ByteString, ByteString, ByteString)
+getAssociativeArgs = \case
+  AssociativeMismatched x y z -> (x, y, z)
+  AssociativeMatched x y z _  -> (x, y, z)
+
+_AssociativeResult :: Fold AssociativeCase ByteString
+_AssociativeResult = folding $ \case
+  AssociativeMatched _ _ _ res -> pure res
+  _                            -> Nothing
+
+associative ::
+  (Word8 -> Word8 -> Word8) ->
+  DefaultFun ->
+  PropertyT IO ()
+associative f b = do
+  testCase <- forAllWith ppShow . genAssociativeCase $ f
+  cover 45 "mismatched lengths" . hasn't _AssociativeResult $ testCase
+  cover 45 "matched lengths" . has _AssociativeResult $ testCase
+  let expectedMay = preview _AssociativeResult testCase
+  let (x, y, z) = getAssociativeArgs testCase
+  outcome <- associatively b x y z
+  case (outcome, expectedMay) of
+    ((EvaluationFailure, EvaluationFailure), Nothing) -> success
+    (_, Nothing) -> annotate "Unexpected failure" >> failure
+    ((EvaluationSuccess leftAssoc, EvaluationSuccess rightAssoc), Just expected) -> do
+      leftAssoc === rightAssoc
+      leftAssoc === mkConstant () expected
+    _ -> annotate "Unexpected failure" >> failure
+
+associatively ::
+  DefaultFun ->
+  ByteString ->
+  ByteString ->
+  ByteString ->
+  PropertyT IO (EvaluationResult (Untyped.Term Name DefaultUni DefaultFun ()),
+                EvaluationResult (Untyped.Term Name DefaultUni DefaultFun ()))
+associatively fun x y z = do
+  let x' = mkConstant @ByteString () x
+  let y' = mkConstant @ByteString () y
+  let z' = mkConstant @ByteString () z
+  let leftAssoc = mkIterApp () (builtin () fun) [
+        mkIterApp () (builtin () fun) [x', y'],
+        z'
+        ]
+  let rightAssoc = mkIterApp () (builtin () fun) [
+        x',
+        mkIterApp () (builtin () fun) [y', z']
+        ]
+  bitraverse cekEval cekEval (leftAssoc, rightAssoc)
+
+self :: DefaultFun -> PropertyT IO ()
+self b = do
+  bs <- forAllWith ppShow . Gen.bytes $ byteBoundRange
+  let bs' = mkConstant @ByteString () bs
+  let comp = mkIterApp () (builtin () b) [bs', bs']
+  outcome <- cekEval comp
+  case outcome of
+    EvaluationSuccess res -> res === mkConstant @ByteString () bs
+    _                     -> failure
+
+data AbsorbingCase =
```

Comment:
Another mystery datatype. And it's different to the others, so I can't even use the knowledge I've already reverse-engineered. You know what I'm going to say :)